src/lib_info.ml

```diff
@@ -268,4 +301,24 @@ let of_dune_lib dp =
   ; special_builtin_support = Lib.special_builtin_support dp
   }
 
-let orig_src_dir t = Option.value ~default:t.src_dir t.orig_src_dir
+type external_ = Path.t t
+type local = Path.Build.t t
+
+let map t ~f_path ~f_obj_dir =
+  let f = f_path in
+  let list = List.map ~f in
+  let mode_list = Mode.Dict.map ~f:list in
+  { t with
+    src_dir = f t.src_dir
+  ; orig_src_dir = Option.map ~f t.orig_src_dir
+  ; obj_dir = f_obj_dir t.obj_dir
+  ; archives = mode_list t.archives
+  ; plugins = mode_list t.plugins
+  ; foreign_objects = Source.map ~f:(List.map ~f) t.foreign_objects
+  ; foreign_archives = mode_list t.foreign_archives
+  ; jsoo_runtime = List.map ~f t.jsoo_runtime
+  ; jsoo_archive = Option.map ~f t.jsoo_archive
+  }
+
+let to_external = map ~f_path:Path.build ~f_obj_dir:Obj_dir.of_local
```

Comment:
We should make the naming more consistent between `Obj_dir` and this module. In `Obj_dir` we have:

- `of_local`: convert build path to generic `Path.t` paths
- `convert_to_external`: convert a local library to an external one by prefixing all paths with a generic `Path.t` path

Here, it seems that we want the first version. So this function should be called `of_local`